### PR TITLE
Created pfTrendsChart directive, ngdocs and unit tests

### PR DESCRIPTION
**Pull request by dtaylor113**

### CODE REVIEW

**Comment by jeff-phillips-18** [comment]
This is really a card. It should it be in the cards module and renamed appropriately.

File: src/charts/trends/trends-chart.directive.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+             <input type="checkbox" ng-model="custShowYAxis">Y Axis</input>
+           </label>
+         </div>
+       </form>
+     </div>
+     <div class="col-md-4">
+       <form role="form" >
+         <div class="form-group">
+           <label>Chart Height</label></br>
+           <input style="height:25px; width:60px;" type="number" ng-model="custChartHeight"></input>
```
Comment (jeff-phillips-18) [inline src/charts/trends/trends-chart.directive.js]: move inline styles to css file (coming soon) or to a style block.

**Comment by erundle** [comment]
I don't think we want to add this to the cards module. That puts a dependency on cards that c3 is included which then add affects the base angular patternfly module.

I would prefer to see the card wrapper around this module be removed and just create a trends directive that could be used inside or outside a card.

File: src/charts/trends/trends-chart.directive.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * <li>.yData  - Array, Y Values for the data points, first element must be the name of the data
+ * </ul>
+ *
+ * @param {int=} chartHeight   height of the sparkline chart
+ * @param {boolean=} showXAxis override sparkline config settings for showing the X Axis
+ * @param {boolean=} showYAxis override sparkline config settings for showing the Y Axis
+ * @example
+ <example module="patternfly.charts">
+ <file name="index.html">
+  <div ng-controller="ChartCtrl" class="row" style="display:inline-block; width: 100%;">
```
Comment (erundle) [inline src/charts/trends/trends-chart.directive.js]: Is this CSS needed?

File: src/charts/trends/trends-chart.directive.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+   </div>
+   </div>
+  </div>
+ </file>
+
+ <file name="script.js">
+ angular.module( 'patternfly.charts' ).controller( 'ChartCtrl', function( $scope ) {
+
+       $scope.config = {
+         'chartId': 'exampleTrendsChart',
```
Comment (erundle) [inline src/charts/trends/trends-chart.directive.js]: I personally don't care how this is spaced but it should probably be consistent :)

File: test/charts/trends/trends-chart.spec.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+describe('Directive: pfTrendsChart', function() {
+  var $scope, $compile, element, isolateScope;
+
+  beforeEach(module(
+    'patternfly.charts',
+    'charts/trends/trends-chart.html',
+    'card/basic/card.html',
+    'charts/sparkline/sparkline-chart.html'
```
Comment (erundle) [inline test/charts/trends/trends-chart.spec.js]: Is sparkline needed here?

File: src/charts/trends/trends-chart.directive.js (hunk 2 above)
Comment (dtaylor113) [inline src/charts/trends/trends-chart.directive.js]: Is what CSS needed?  ..the <ul>?

**Comment by waldenraines** [comment]
> Ugh...wish I saw this before I changed everything ;-(
> Let me digest this suggestion
> -DT

You could always just remove the third commit using `git rebase` to get back to where you were.

**Comment by dtaylor113** [comment]
I don't think we want to add this to the cards module. That puts a dependency on cards that c3 is   
  included which then add affects the base angular patternfly module.

I suggest we leave it as a chart for now as we don't want to put the c3 dependency on cards.

  I would prefer to see the card wrapper around this module be removed and just create a trends 
  directive that could be used inside or outside a card.

I need some more information on how you envision this.
Thanks,
- Dave

**Comment by erundle** [comment]
I would remove the outer card wrapper and just make this a trends directive. If this is really needed as a card then theoretically you would have a second directive that injected and wrapped the preexisting trends directive.

**Comment by dtaylor113** [comment]
Removed the outer card wrapper from the trends directive.
Also fixed indentation in ngdocs.

File: src/charts/trends/trends-chart.directive.js (hunk 3 above)
Comment (dtaylor113) [inline src/charts/trends/trends-chart.directive.js]: Fixed

File: src/charts/charts.module.js [hunk 5]
```diff
@@ -5,5 +5,5 @@
  *   Charts module for patternfly. Must Include d3.js and c3.js to use
  *
  */
-angular.module('patternfly.charts', []);
+angular.module('patternfly.charts', ['patternfly.card']);
```
Comment (erundle) [inline src/charts/charts.module.js]: Can this dependency be removed now?

File: src/charts/trends/trends-chart.directive.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.charts.directive:pfTrendsChart
+ *
+ * @description
+ *   Directive for rendering a trend chart. The trend chart combines overall data with a pfCard and a
+ *   pfSparklineChart.
```
Comment (erundle) [inline src/charts/trends/trends-chart.directive.js]: Wording update needed.

File: src/charts/charts.module.js (hunk 5 above)
Comment (dtaylor113) [inline src/charts/charts.module.js]: Removed

File: src/charts/trends/trends-chart.directive.js (hunk 6 above)
Comment (dtaylor113) [inline src/charts/trends/trends-chart.directive.js]: Fixed

**Comment by dtaylor113** [comment]
Ok, I believe I have addressed all issues.  Can this be merged?
Thanks,
- Dave

**Comment by erundle** [comment]
LGTM

**Pull request merged**